File: Easy/Check if a number is divisible by 8/check-if-a-number-is-divisible-by-8.js
//{ Driver Code Starts
//Initial Template for javascript

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++)
    {
        let S = readLine().trim();
        let obj = new Solution();
        let res = obj.DivisibleByEight(S);
        console.log(res);
    }
}

// } Driver Code Ends


//User function Template for javascript
/**
 * @param {string} s
 * @returns {boolean}
*/

class Solution {
    DivisibleByEight(s) {
        return parseInt(s.substring(Math.max(0, s.length - 3))) % 8 == 0 ? 1 : -1;
    }
}

File: Easy/Check if a number is divisible by 8/check-if-a-number-is-divisible-by-8.py
#User function Template for python3

class Solution:
    def DivisibleByEight(self, s: str) -> int:
        return 1 if int(s[max(0, len(s) - 3):]) % 8 == 0 else -1


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__=='__main__':
    t=int(input())
    for _ in range(t):
        S=input()
        ob=Solution()
        print(ob.DivisibleByEight(S))
# } Driver Code Ends